### DISCUSSION TITLE
Initial load Performance: FadeDuration

### DISCUSSION THREAD

**Discussion by pramilk** [maplibre/maplibre-gl-js#2443] [Ideas]
Currently map uses a default fadeDuration of 300ms. Its a additional 450-600ms for initial load to complete even though the default is 300ms. 

About [fadeDuration](https://maplibre.org/maplibre-gl-js-docs/api/map/): Controls the duration of the fade-in/fade-out animation for label collisions, in milliseconds. This setting affects all symbol layers. This setting does not affect the duration of runtime styling transitions or raster tile cross-fading. Default 300ms"

Proposal:
Update it to be 0 ms on initial load and then on first idle event, change it to the default value (or whatever user has provided). This is a breaking change in experience for performance. This will provide fast initial loading but will still preserve the current fading experience as user zoom around.

Here is with and without change:

- Current (@3.0.0-pre.5): https://perfmeasure.blob.core.windows.net/$web/MapLibre.html
![image](https://user-images.githubusercontent.com/15951646/234094298-0d8a495e-e961-43ab-bf87-6b0768a57d3c.png)
- With proposed change(@3.0.0-pre.5): https://perfmeasure.blob.core.windows.net/$web/MapLibre_FadeChanges.html
![image](https://user-images.githubusercontent.com/15951646/234094427-9f3d8b9e-cfbe-4791-854c-36e83a744231.png)
- Note about metrics
    - Render Complete: It is the time from navigation start till idle event.

**Comment by HarelM** [comment]
Looks good to me. Adds some complexity probably, but seems needed, non the less.

**Reply by pramilk** [reply]
PR: https://github.com/maplibre/maplibre-gl-js/pull/2447
